### PR TITLE
Fix IDE0001 / IDE0002 (simplify names) in corelib

### PR DESCRIPTION
**Pull request by stephentoub**
Applied fixers (and then a few manual changes for things under ifdef)

### CODE REVIEW

**Comment by ShreyasJejurkar** [comment]
Maybe for those rules, we can add a tweak severity level to the warning using `.editorconfig`, so that it would get enforced to the new code as well! So that we can avoid this jumbo cleanup in the future!

File: src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs [hunk 1]
```diff
@@ -640,8 +641,8 @@ internal enum EnableNoGCRegionCallbackStatus
         /// Register a callback to be invoked when we allocated a certain amount of memory in the no GC region.
         /// <param name="totalSize">The total size of the no GC region. Must be a number > 0 or an ArgumentOutOfRangeException will be thrown.</param>
         /// <param name="callback">The callback to be executed when we allocated a certain amount of memory in the no GC region..</param>
-        /// <exception cref="System.ArgumentOutOfRangeException"> The <paramref name="totalSize"/> argument is less than or equal to 0.</exception>
-        /// <exception cref="System.ArgumentNullException">The <paramref name="callback"/> argument is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"> The <paramref name="totalSize"/> argument is less than or equal to 0.</exception>
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs:644]: Is it ok to omit the namespace in doc comments?

Comment (stephentoub) [reply src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs:644]: Yup

Comment (jkotas) [reply src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs:644]: Looks like a difference between https://github.com/dotnet/dotnet-api-docs conventions and the inline docs conventions. I do not think it is ok to omit namespace in dotnet-api-docs.

**Review by jkotas — APPROVED** [review]

**Pull request merged**